File: backend/app/routers/alerts.py
import uuid
from datetime import datetime

from fastapi import APIRouter, HTTPException, Query

from app.mock_store import alerts
from app.schemas.alert import AlertWithLogs

router = APIRouter(prefix="/alerts", tags=["alerts"])


@router.get("/", response_model=list[AlertWithLogs])
async def list_alerts(
    limit: int = Query(50, ge=1, le=200),
    offset: int = Query(0, ge=0),
    start_date: datetime | None = Query(None),
    end_date: datetime | None = Query(None),
):
    items = sorted(alerts.values(), key=lambda x: x["triggered_at"], reverse=True)
    if start_date:
        items = [a for a in items if a["triggered_at"] >= start_date]
    if end_date:
        items = [a for a in items if a["triggered_at"] <= end_date]
    return items[offset : offset + limit]


@router.get("/{alert_id}", response_model=AlertWithLogs)
async def get_alert(alert_id: uuid.UUID):
    alert = alerts.get(alert_id)
    if not alert:
        raise HTTPException(status_code=404, detail="Alert not found")
    return alert

File: backend/src/routers/alerts.js
'use strict';

const { Router } = require('express');
const { alerts } = require('../mockStore');

const router = Router();

// GET /api/alerts?limit=50&offset=0&start_date=...&end_date=...
router.get('/', (req, res) => {
  const limit = Math.min(parseInt(req.query.limit || '50', 10), 200);
  const offset = Math.max(parseInt(req.query.offset || '0', 10), 0);
  const startDate = req.query.start_date ? new Date(req.query.start_date) : null;
  const endDate = req.query.end_date ? new Date(req.query.end_date) : null;

  let items = [...alerts.values()].sort((a, b) => b.triggered_at.localeCompare(a.triggered_at));
  if (startDate) items = items.filter((a) => new Date(a.triggered_at) >= startDate);
  if (endDate) items = items.filter((a) => new Date(a.triggered_at) <= endDate);
  res.json(items.slice(offset, offset + limit));
});

// GET /api/alerts/:id
router.get('/:id', (req, res) => {
  const alert = alerts.get(req.params.id);
  if (!alert) return res.status(404).json({ detail: 'Alert not found' });
  res.json(alert);
});

module.exports = router;
